src/providers/twitch/eventsub/Controller.cpp

```diff
@@ -81,6 +86,12 @@ Controller::~Controller()
 
     this->work.reset();
 
+    if (!this->stoppedFlag.waitFor(std::chrono::milliseconds(250)))
+    {
+        qCWarning(LOG) << "Force end";
+        return;
```

Comment:
This would call `std::terminate` if the thread isn't joined when it's destructed. Not sure if that's what we want. Probably depends on what else would happen afterwards.